src/bmp.imageio/bmpinput.cpp

```diff
@@ -269,4 +281,18 @@ BmpInput::read_color_table(void)
     return true;  // ok
 }
 
+bool
+BmpInput::color_table_is_all_gray(void)
+{
+    size_t ncolors = m_colortable.size();
+    for (size_t i = 0; i < ncolors; i++) {
+        color_table& color = m_colortable[i];
+        if ((color.b == color.g) && (color.g == color.r))
+            continue;
+        else
+            return false;
```

Comment:
Since the `continue` is exactly what happens at the end of the loop anyway, can/should this be simplified to:

    if (color.r != color.g || color.r != color.b)
        return false;

Reply:
I applied this simplification, which is better than my initial verbose implementation.